### PR TITLE
MZ harp 13 - Updated gulpfile.js & dependecies on package.json

### LINKED ISSUE #13: Fixing gulp file

### PR DESCRIPTION
**Pull request by pdsdesign**
Closes #13

### CODE REVIEW

**Comment by pdsdesign** [comment]
@lmcorreia Could you please give it a look and review it?

File: gulpfile.js [hunk 1]
```diff
@@ -1,3 +1,4 @@
+var _           = require('lodash');
```
Comment (lmcorreia) [inline gulpfile.js]: Why do we need `lodash`?

Comment (pdsdesign) [inline gulpfile.js]: It must have come from a previous version, I'm going to remove it from gulp.js and its dependency on package.json

File: gulpfile.js [hunk 2]
```diff
@@ -7,11 +8,13 @@ var harpConfig  = require('./harp.json');
 var jimp        = require('gulp-jimp');
 var concat      = require('gulp-concat');
 var uglify      = require('gulp-uglify');
+var runSequence = require('run-sequence');
+
```
Comment (lmcorreia) [inline gulpfile.js:11]: If we ran this through JSLint, I'm pretty sure we'd get a "too many var declarations", we only need one:

```
var gulp = require('gulp'),
    browserSync = require('browser-sync'),
[...]
    uglify = require('gulp-uglify'),
    runSequence = require('run-sequence');
```

Also, introducing extra spaces between var name and "=" operator is not standard practice and IMHO doesn't make for an easier read.

File: gulpfile.js [hunk 3]
```diff
@@ -51,77 +61,135 @@ gulp.task('serve', function (done) {
...
+
 /**
  * Build the Harp Site
  */
-gulp.task('build', ['jimp', 'uglify'], function (done) {
-  cp.exec('harp compile . www', {stdio: 'inherit'})
-    .on('close', done);
+gulp.task('build', ['copy-css-assets', 'jimp', 'uglify'], function () {
+  return cp.exec('harp compile . www', {stdio: 'inherit'})
```
Comment (lmcorreia) [inline gulpfile.js]: Missing semicolon (`;`) at the end

File: gulpfile.js [hunk 4]
```diff
@@ -51,77 +61,135 @@ gulp.task('serve', function (done) {
...
+    imgDest         = 'public/images/posts/',
+    imgQuality      = 80,
+    largeWidth      = harpConfig.globals.breakpoints.large,
+    regularWidth    = harpConfig.globals.breakpoints.regular,
+    mediumWidth     = harpConfig.globals.breakpoints.medium,
+    smallWidth      = harpConfig.globals.breakpoints.small
+
+// Clean the image folder
+gulp.task('jimp-clean', function() {
+  return cp.exec('rm ' + imgDest + '*', {stdio: 'inherit'});
```
Comment (lmcorreia) [inline gulpfile.js]: Missing semicolon (`;`) at the end

File: gulpfile.js [hunk 5]
```diff
@@ -51,77 +61,135 @@ gulp.task('serve', function (done) {
...
+
+
+// Jimp variables
+var imgSrc          = 'public/_posts-images/*',
+    imgDest         = 'public/images/posts/',
+    imgQuality      = 80,
+    largeWidth      = harpConfig.globals.breakpoints.large,
+    regularWidth    = harpConfig.globals.breakpoints.regular,
+    mediumWidth     = harpConfig.globals.breakpoints.medium,
+    smallWidth      = harpConfig.globals.breakpoints.small
```
Comment (lmcorreia) [inline gulpfile.js]: Missing semicolon (`;`) at the end

Comment (lmcorreia) [inline gulpfile.js]: We should not introduce extra spaces (see var declarations up top).

We also need to refactor this a bit. We cannot be declaring variables in multiple places. JS code standards say that you declare all the variables you're going to use at the top of your function - first line. At the very least we need to move this to the top of the file but the best thing would be to only declare each one at the top of the functions it is used in. Alternatively, we could try to wrap this whole thing in a function and put these variables there.

**Comment by jolidog** [comment]
@lmcorreia going to merge this since we want to fork with it in place.

**Pull request merged**